paimon-core/src/main/java/org/apache/paimon/mergetree/MergeTreeWriter.java

```diff
@@ -133,6 +139,23 @@ public MergeTreeWriter(
             compactChangelog.addAll(increment.compactIncrement().changelogFiles());
             updateCompactDeletionFile(increment.compactDeletionFile());
         }
+
+        // TODO: Verify the performance of the default implementation of copy,
+        //  and remove the following code block if proved to be of no regression.
+        boolean isCopyMethodOverridden;
+        try {
+            isCopyMethodOverridden =
```

Comment:
Can we just introduce a new FileIO interface?
```
interface NativeCopyFileIO extends FileIO {
    void nativeCopy(Path src, Path dst);
}
```

Reply:
According to offline discussion, I benchmarked the throughput of the default implementation introduced in FileIO in Aliyun, and it seems that the default implementation has been enough to bring performance improvement. Given this result, I'll directly add the copy method to FileIO, instead of introducing the new NativeCopyFileIO interface.